src/ray/raylet/node_manager.cc

```diff
@@ -414,43 +404,56 @@ void NodeManager::Heartbeat() {
           ResourceSet(local_resources.GetTotalResources()));
     }
 
+    if (!last_heartbeat_resources_.GetAvailableResources().IsEqual(
+            local_resources.GetAvailableResources())) {
+      if (local_resources.GetAvailableResources().IsEmpty()) {
+        (*heartbeat_data->mutable_resources_available())[kResourcesTurningEmpty] = 0.0;
+      } else {
+        for (const auto &resource_pair :
+             local_resources.GetAvailableResources().GetResourceMap()) {
+          (*heartbeat_data->mutable_resources_available())[resource_pair.first] =
+              resource_pair.second;
+        }
+      }
+      last_heartbeat_resources_.SetAvailableResources(
+          ResourceSet(local_resources.GetAvailableResources()));
+    }
+
     local_resources.SetLoadResources(local_queues_.GetTotalResourceLoad());
     if (!last_heartbeat_resources_.GetLoadResources().IsEqual(
             local_resources.GetLoadResources())) {
-      for (const auto &resource_pair :
-           local_resources.GetLoadResources().GetResourceMap()) {
-        (*heartbeat_data->mutable_resource_load())[resource_pair.first] =
-            resource_pair.second;
+      if (local_resources.GetLoadResources().IsEmpty()) {
+        (*heartbeat_data->mutable_resource_load())[kResourcesTurningEmpty] = 0.0;
+      } else {
+        for (const auto &resource_pair :
+             local_resources.GetLoadResources().GetResourceMap()) {
+          (*heartbeat_data->mutable_resource_load())[resource_pair.first] =
+              resource_pair.second;
+        }
       }
       last_heartbeat_resources_.SetLoadResources(
           ResourceSet(local_resources.GetLoadResources()));
     }
   } else {
     // If light heartbeat disabled, we send whole resources information every time.
     for (const auto &resource_pair :
-         local_resources.GetAvailableResources().GetResourceMap()) {
-      (*heartbeat_data->mutable_resources_available())[resource_pair.first] =
+         local_resources.GetTotalResources().GetResourceMap()) {
```

Comment:
Why did you switch these two?

Reply:
in elsewhere the codes normal handle total resources first, then available and load. I'd like keep them same order for readability.